### DISCUSSION TITLE
configuring Budgets in AWS

### DISCUSSION THREAD

**Discussion by ostapkostyk** [arendst/Tasmota#13050] [Q&A] [answered]
Hi,
I've configured MQTT broker at AWS using this article: https://tasmota.github.io/docs/AWS-IoT/ . It works fine, I get MQTT messages from my ESP8266. 
Now I want to set up Budgets in the AWS to protect an account from unexpected costs but can't figure out how to do it.
In IAM I have two tasmota-related roles:
TasmotaAuth-LambdaBasicExecRoleRandomGenerator-xxxxxxxxx
TasmotaAuth-LambdaBasicExecRoleRetain-xxxxxxxxxxx

I've created a Budget, configured an Alert, and trying to add an action. For this I have to apply some Policy. Shall I create some policy with "Deny" action on IoT service, or?

**Comment by s-hadinger** [comment, marked as answer]
Keep in mind that AWS IoT pricing is described here: https://aws.amazon.com/iot-core/pricing/ and will very likely be zero if you have a small number of devices.

**Comment by ostapkostyk** [comment]
thanks for all answers!
I've decided to discontinue this project and close the AWS account.
The reason is a danger of getting an enormous bill in case if an account is hacked or just an unexpectedly high bill in case of own mistake (and it looks quite complex to set their budget events in a right way, there is even warning that it will not work if bla-bla-bla).
I'm just doing DYI project, not a business, so don't want to take such a big risk, even if the probability is low and despite I use 2fa.
I'll try to set-up an MQTT broker and related services on fixed price hosting.
I've been inspired to act by this: https://www.quora.com/My-AWS-account-was-hacked-and-I-have-a-50-000-bill-how-can-I-reduce-the-amount-I-need-to-pay

**Reply by sfromis** [reply]
For limited use, you can also find free public MQTT brokers on the internet.

**Reply by Jason2866** [reply]
Setting up your own local mqtt server is VERY easy and cheap to do. Use a RasPi or maybe you have a device running you can install on (NAS, AccessPoint....)
Search for `mosquitto`

**Reply by barbudor** [reply]
I made some tests with Flespi and MyQTTHub
Flepsi had my preference
Easy to configure when you know about MQTT but from a First Starter, installing Mosquitto on a PC/RPi is very easy

**Reply by ostapkostyk** [reply]
My project is to log temp/humidity on a remote site, 4-10 sensors. My idea was to use AWS to save on RasPi. My sensors are quite cheap since it's bare-metal, so at the end, it would cost me just 20-40 USD for the complete project if I use AWS for free (at least the first year).
Now I'm thinking to install RasPi. The other design issues are logging and visualizing the data. 
Currently, I try to use openHAB with Mosquitto MQTT. Then I'll probably try to install InfluxDB and write some Python script to push the data. If I'll not manage with InfluxDB then I'll just write to the text file and interpret results on my laptop.
My other idea was to host Mosquitto in a cloud on some cheap hosting service (to avoid buying RPi) + InfluxDB (or any else time-series-optimized DB) + Grafana + my Python script to glue all this together. But this might be too long way for a simple project. Time is very limited resource in this project.
So, regarding free public MQTT - not sure what can I do with it if I have to log the data. Maybe just run my MQTT client on my own server to query this public broker and then saving the measurements on my server... then I could save some headache related to setting up a MQTT broker... maybe.
Will look also on MyQttHub and Flespi, thanks for the reference!

**Reply by barbudor** [reply]
There's also the AdafruitIO but it requires some very specific MQTT topics so not easy to configure tasmota with it and leverage all capabilities

**Reply by ostapkostyk** [reply]
Flespi looks promising. I'd run some docker container in a cloud with Python script and database to grab data out of Flespi using their stream.
Just one question - is it easy to read temperature/humidity from tasmota on that platform? I mean, as far as I know, tasmota doesn't provide separate values per MQTT topic, but JSON string by topic ..../SENSOR. In AWS one can use Lambda to parse JSON for particular values. How about Flespi? 
P.S. I use AM2302 sensor

**Reply by barbudor** [reply]
If you are about to use docker for some script, then you don't have much to do to also run mosquitto in a docker...

Mosquitto or Flepsi are brokers. They relay messages as their are. They don't process data.
So it's up to you script to parse the Tasmota SENSOR message and split values to write in your database.

You may also  look at InfluxDB or Prometeus.
Tasmota has drivers for both which allows to publish data without going through MQTT
Grafana can then be used to build graphs
